testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/security/realms/AddSecretAuthenticationOnlineTest.java

```diff
@@ -0,0 +1,103 @@
+    private static final Address TEST_REALM_ADDRESS
+            = Address.coreService("management").and("security-realm", TEST_REALM_NAME);
+    private static final Address TEST_SECRET_IDENTITY_ADDRESS
+            = TEST_REALM_ADDRESS.and("server-identity", "secret");
+    private static final String PASSWORD = "password1";
+
+    private OnlineManagementClient client;
+    private Operations ops;
+    private Administration administration;
+
+    @Before
+    public void connect() throws Exception {
+        client = ManagementClient.online(OnlineOptions.standalone().localDefault().build());
+        ops = new Operations(client);
+        administration = new Administration(client);
+
+        AddSecurityRealm addSecurityRealm = new AddSecurityRealm.Builder(TEST_REALM_NAME).build();
+        client.apply(addSecurityRealm);
+        assertTrue("The security realm should be created", ops.exists(TEST_REALM_ADDRESS));
+    }
+
+    @Test
+    public void add() throws Exception {
+        client.apply(new AddSecretAuthentication.Builder(TEST_REALM_NAME)
+                .password(PASSWORD)
+                .build());
+        assertTrue("Server identity should be created", ops.exists(TEST_SECRET_IDENTITY_ADDRESS));
+        ModelNodeResult result = ops.readAttribute(TEST_SECRET_IDENTITY_ADDRESS, "value");
+        result.assertSuccess();
+        assertEquals("Password should be encoded properly.",
+                PASSWORD, new String(BaseEncoding.base64().decode(result.stringValue())));
+    }
+
+    @Test(expected = CommandFailedException.class)
+    public void addExisting() throws Exception {
+        client.apply(new AddSecretAuthentication.Builder(TEST_REALM_NAME)
+                .password(PASSWORD)
+                .build());
+        assertTrue("Server identyty should be created", ops.exists(TEST_SECRET_IDENTITY_ADDRESS));
+        client.apply(new AddSecretAuthentication.Builder(TEST_REALM_NAME)
+                .password(PASSWORD)
+                .build());
+        fail("Secret authentication method has been already configured. Should have failed.");
+    }
+
+    @Test
+    public void replaceExisting() throws Exception {
+        client.apply(new AddSecretAuthentication.Builder(TEST_REALM_NAME)
+                .password("secret")
+                .build());
+        assertTrue("Server identyty should be created", ops.exists(TEST_SECRET_IDENTITY_ADDRESS));
+
+        client.apply(new AddSecretAuthentication.Builder(TEST_REALM_NAME)
+                .password(PASSWORD)
+                .replaceExisting()
+                .build());
+        assertTrue("Server identity should be created", ops.exists(TEST_SECRET_IDENTITY_ADDRESS));
+        ModelNodeResult result = ops.readAttribute(TEST_SECRET_IDENTITY_ADDRESS, "value");
+        result.assertSuccess();
+        assertEquals("Password should be encoded properly.",
+                PASSWORD, new String(BaseEncoding.base64().decode(result.stringValue())));
```

Comment:
Never ever do `new String(byte[])`, that uses the default charset. Please use `new String(byte[], String|Charset)`.